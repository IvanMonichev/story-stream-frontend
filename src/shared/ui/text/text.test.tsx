import { render, screen } from '@testing-library/react'
import { describe, expect, it } from 'vitest'
import Text from './text'

describe('Text component', () => {
  it('renders children correctly in Paragraph', () => {
    render(<Text.Paragraph>Test Paragraph</Text.Paragraph>)
    const paragraph = screen.getByText('Test Paragraph')
    expect(paragraph).toBeInTheDocument()
  })

  it('does not apply intrigued class when intrigued prop is false', () => {
    render(<Text.Paragraph intrigued={false}>Test Paragraph</Text.Paragraph>)
    const paragraph = screen.getByText('Test Paragraph')
    expect(paragraph).not.toHaveClass('textIntrigue')
  })

  it('applies custom className', () => {
    render(
      <Text.Paragraph className='custom-class'>Test Paragraph</Text.Paragraph>
    )
    const paragraph = screen.getByText('Test Paragraph')
    expect(paragraph).toHaveClass('custom-class')
  })

  it('matches snapshot for Paragraph', () => {
    const { asFragment } = render(
      <Text.Paragraph>Test Paragraph</Text.Paragraph>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders children correctly in Text component', () => {
    render(<Text>Some Text</Text>)
    expect(screen.getByText('Some Text')).toBeInTheDocument()
  })
})
