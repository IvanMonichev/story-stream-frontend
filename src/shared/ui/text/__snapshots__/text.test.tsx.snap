// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Text component > matches snapshot for Paragraph 1`] = `
<DocumentFragment>
  <p
    class="_text_435960"
  >
    Test Paragraph
  </p>
</DocumentFragment>
`;
